### PR TITLE
Issue#2849: added logic to check ifff version

### PR DESCRIPTION
**Pull request by yingfeng-iu**
Issue#2849: added tests for ifff version

Issue#2849: updated tests

Issue#2849: added tests for ifff version

Issue#2849: change IFFF default version to constant

Issue#2849: fix iiif typo

Issue#2849: improve IFFF constants

Issue#2849: improve tests to be more flexible

Issue#2849: fix a wrong reference to constant

Issue#2849: add clean db in tests

Fixes #issuenumber ; refs #issuenumber

Present tense short summary (50 characters or less)

More detailed description, if necessary. Try to be as descriptive as you can: even if you think that the PR content is obvious, it may not be obvious to others. Include tracebacks if helpful, and be sure to call out any bits of the PR that may be work-in-progress.

Description can have multiple paragraphs and you can use code examples inside:

``` ruby
class PostsController
  def index
    respond_with Post.limit(10)
  end
end
```

Changes proposed in this pull request:
*
*
*

@samvera-labs/avalon

### CODE REVIEW

File: spec/controllers/hyrax/generic_works_controller_spec.rb [hunk 1]
```diff
@@ -2,8 +2,93 @@
...
+  routes { Rails.application.routes }
+  let(:main_app) { Rails.application.routes.url_helpers }
+  let(:hyrax) { Hyrax::Engine.routes.url_helpers }
+  let(:user) { create(:user) }
+
+  before { sign_in user }
+
+  describe '#manifest' do
+    let(:work) { create(:work_with_one_file, user: user) }
+    let(:file_set) { work.ordered_members.to_a.first }
```
Comment (cjcolvar) [inline spec/controllers/hyrax/generic_works_controller_spec.rb]: Since the tests mock out the manifest factory, I think you could mock the work show presenter and not need to create a work, file set, or file objects.  This would speed up the tests a lot and avoid needing to add the fixture files to the repo.  And I think the :clean_repo could be dropped as well.

You can mock methods on the controller using the `controller` object:
`allow(controller).to receive(:presenter).and_return(presenter)`

File: app/controllers/hyrax/generic_works_controller.rb [hunk 2]
```diff
@@ -6,19 +6,50 @@ class GenericWorksController < ApplicationController
...
+    IIIF_DEFAULT_VERSION = 2
+    IIIF_DEFAULT_MANIFEST_MIME = "application/json;profile=http://iiif.io/api/presentation/#{IIIF_DEFAULT_VERSION}/context.json".freeze
+
     self.curation_concern_type = ::GenericWork
 
     # Use this line if you want to use a custom presenter
     self.show_presenter = Hyrax::GenericWorkPresenter
 
+    # @return the highest IIIF version (as an integer) specified in the Accept request header, or the default version if none specified
+    def iiif_version
```
[the remark this answers is not in the document]
Comment (yingfeng-iu) [reply app/controllers/hyrax/generic_works_controller.rb]: Sure. Ideally I was hoping to put these into some service/util class so it could be useful outside the controller.

**Comment by yingfeng-iu** [comment]
I found a small bug in the iiif_version method, I've fixed it and adding more tests to cover all corner cases.

**Comment by yingfeng-iu** [comment]
I've run all 4 cases (one test for each) with default version set to either 2 or 3, and they all passed. if code climate passes, then the last commit is ready for review again.

File: spec/controllers/hyrax/generic_works_controller_spec.rb [hunk 3]
```diff
@@ -3,7 +3,73 @@
...
+      end
+    end
+
+    context 'for request without IIIF profile' do
+      let(:mime) { "application/json;" }
+
+      it 'returns IIIF default version manifest' do
+        get :manifest, params: { id: 'testwork', format: :json }
+        expect(response.headers['Content-Type']).to eq Hyrax::GenericWorksController::IIIF_DEFAULT_MANIFEST_MIME
+        if Hyrax::GenericWorksController::IIIF_DEFAULT_VERSION == 2
```
Comment (cjcolvar) [inline spec/controllers/hyrax/generic_works_controller_spec.rb]: This conditional makes the test resilient to the change of the `IIIF_DEFAULT_VERSION` from 2 to 3, but since the controller code is set to 2 I'd expect this test to only cover the Presentation 2 case and when that constant changes then this test would change too.
